dashboard-app/src/features/dashboard/Dashboard/index.tsx

```diff
@@ -25,46 +34,57 @@ const Dashboard: React.FunctionComponent<RouteComponentProps> = (props) => {
         </Col>
       </Row>
       <Row center="xs">
-        <Col xs={8}>
-          <Row>
-            <Col xs={6}>
-              <Polaroid
-                title={TitlesCopy.Time.title}
-                caption={TitlesCopy.Time.subtitle}
-                callToAction="View data"
-                picture="T"
-                onClick={() => Pages.navigateTo('Time', props.history.push)}
-              />
-            </Col>
-            <Col xs={6}>
-              <Polaroid
-                title={TitlesCopy.Activities.title}
-                caption={TitlesCopy.Activities.subtitle}
-                callToAction="View data"
-                picture="A"
-                onClick={() => Pages.navigateTo('Activity', props.history.push)}
-              />
-            </Col>
-            <Col xs={6}>
-              <Polaroid
-                title={TitlesCopy.Volunteers.title}
-                caption={TitlesCopy.Volunteers.subtitle}
-                callToAction="View data"
-                picture="V"
-                onClick={() => Pages.navigateTo('Volunteer', props.history.push)}
-              />
-            </Col>
-            <Col xs={6}>
-              <Polaroid
-                title="Projects"
-                caption="See what projects volunteers spend their time doing."
-                callToAction="Coming soon"
-                picture="P"
-                disabled
-              />
-            </Col>
-          </Row>
-        </Col>
+        <LoadingBoundary isLoading={loading}>
+          <ThrowIfError error={error} />
```

Comment:
This is kind of awkward. I've just imagined a solution that looks like this:

```ts
() => {
  return (
    <LoadingBoundary state={useDashboardStatistics()}>
		{
			(data) => (
				<Col>
					// ...
				</Col>
			)
		}
	</LoadingBoundary>
  );
}
```

In this example, `LoadingBoundary` takes this standard(ish) `{ loading, error, data }` object, and takes care of rendering loaders and error views if either `loading` or `error` are truthy. Once `data` is available, it's passed as a prop to the child component, which is rendered.

But tbh I don't know if we want to do this now.

Reply:
This is actually a bad idea because it breaks one of the rules of hooks: that they should be called at the top level. But I pushed something that does something similar.